### ISSUE TITLE
Support reading/writing COSE_Key values

### ISSUE THREAD

**Issue by bartonjs** [dotnet/runtime#32080]
COSE_Key comes from [IETF RFC 8152, section 7](https://tools.ietf.org/html/rfc8152#section-7).

Some specs are already known that use COSE Key encodings that don't use all of COSE, so at a high level we should support going from `AsymmetricAlgorithm` to an encoded COSE_Key (`byte[]` / write to a `Span<byte>`), and going from a COSE_Key to `AsymmetricAlgorithm` types.

In order to maintain type safety, and reduce the risk of over-casting by callers, it probably should be something like

```C#
partial class CoseKey
{
    public CoseKeyType KeyType { get; }
    // For symmetry with certs, returns null when the key type isn't an RSA type.
    public RSA? GetRSAKey() => throw null;
    public static bool TryParseRSAKey(ReadOnlySpan<byte> encodedKey, out RSA key) => throw null;
    public static byte[] EncodeKey(RSA key);
}
```

Repeat for ECDsa and ECDiffieHellman.  Technically it could also be defined for HMAC and symmetric key wrap, but it seems less valuable.

(Reading is probably needed before writing)

**Comment by vcsjones** [comment]
@bartonjs out of curiosity, what happened to the COSE epic?

**Comment by bartonjs** [comment]
We never had an item for an "epic", but the project board is still there.  But, assuming you mean pragmatically instead of paperwork-related: The manpower funding for COSE was based on it being a dependency of FIDO2.  But as we got into the FIDO2 work we saw that it doesn't use COSE other than serializing keys as COSE_Key.  So we made the CBOR primitives, proved that COSE_Key values can be built from our crypto keys (and vice versa) and at that point we no longer had any user to validate the usability/completeness of a COSE implementation.
